cloudflare/resource_cloudflare_custom_ssl_test.go

```diff
@@ -0,0 +1,104 @@
+package cloudflare
+
+import (
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"os"
+	"regexp"
+	"testing"
+)
+
+func TestAccCloudflareCustomSSL_Basic(t *testing.T) {
+	t.Parallel()
+	var customSSL cloudflare.ZoneCustomSSL
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	resourceName := "cloudflare_custom_ssl." + rnd
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareCustomSSLDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareCustomSSLCertBasic(zoneID, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareCustomSSLExists(resourceName, &customSSL),
+					resource.TestCheckResourceAttr(
+						resourceName, "zone_id", zoneID),
+					resource.TestMatchResourceAttr(
+						resourceName, "priority", regexp.MustCompile("^[0-9]\\d*$")),
+					resource.TestCheckResourceAttr(
+						resourceName, "status", "active"),
+					resource.TestMatchResourceAttr(
+						resourceName, "zone_id", regexp.MustCompile("^[a-z0-9]{32}$")),
+					resource.TestCheckResourceAttr(
+						resourceName, "custom_ssl_options.bundle_method", "ubiquitous"),
+					resource.TestCheckResourceAttr(
+						resourceName, "custom_ssl_options.type", "legacy_custom"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCloudflareCustomSSLCertBasic(zoneID string, rName string) string {
+	return fmt.Sprintf(`
+resource "cloudflare_custom_ssl" "%[2]s" {
+  zone_id = "%[1]s"
+  options = {
```

Comment:
This is causing a failure for the integration tests

```

------- Stdout: -------
=== RUN   TestAccCloudflareCustomSSL_Basic
=== PAUSE TestAccCloudflareCustomSSL_Basic
=== CONT  TestAccCloudflareCustomSSL_Basic
--- FAIL: TestAccCloudflareCustomSSL_Basic (0.02s)
    testing.go:568: Step 0 error: config is invalid: Unsupported argument: An argument named "options" is not expected here.
FAIL
```

```suggestion
  custom_ssl_options = {
```

Reply:
Odd that this was not picked up by CI. Good catch! I fixed it.

Reply:
We have two types of CI running here and only one hooked up to GitHub 🙁 The unit tests passed but these were making the integration tests fail which run on a different system.

We're looking to see if we can have them both hooked into GitHub but it's a chat with Hashicorp as they own the repositories.